### PR TITLE
Make metrics collection interval configurable

### PR DESCRIPTION
**Pull request by cangencer**
With a default of 5 seconds

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/JetMetricsService.java [hunk 1]
```diff
@@ -77,7 +80,7 @@ public void init(NodeEngine nodeEngine, Properties properties) {
                 lastSize[0] = blob.length;
                 metricsJournal.add(entry(System.currentTimeMillis(), blob));
                 logFine(logger, "Collected %,d metrics, %,d bytes", renderer.getCount(), blob.length);
-            }, COLLECTION_INTERVAL_SECONDS, COLLECTION_INTERVAL_SECONDS, TimeUnit.SECONDS);
+            }, config.getCollectionIntervalSeconds(), config.getCollectionIntervalSeconds(), TimeUnit.SECONDS);
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/JetMetricsService.java]: Initial delay can remain 1000 ms i think

File: hazelcast-jet-core/src/main/resources/hazelcast-jet-default.xml [hunk 2]
```diff
@@ -32,12 +32,14 @@
         <!-- receive window size multiplier, only applies to distributed edges -->
        <receive-window-multiplier>3</receive-window-multiplier>
     </edge-defaults>
+
     <metrics>
-        <!-- if metrics collection is enabled -->
-        <enabled>true</enabled>
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/resources/hazelcast-jet-default.xml]: Should have been turned to an attribute.

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/config/XmlConfigTest.java [hunk 3]
```diff
@@ -57,7 +61,12 @@ public void when_noConfigSpecified_usesDefaultConfig() {
         assertEquals(Runtime.getRuntime().availableProcessors(),
                 jetConfig.getInstanceConfig().getCooperativeThreadCount());
         assertEquals(DEFAULT_FLOW_CONTROL_PERIOD_MS, jetConfig.getInstanceConfig().getFlowControlPeriodMs());
+        assertEquals(DEFAULT_FLOW_CONTROL_PERIOD_MS, jetConfig.getInstanceConfig().getFlowControlPeriodMs());
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/config/XmlConfigTest.java]: duplicated line?

File: hazelcast-jet-core/src/main/resources/hazelcast-jet-default.xml (hunk 2 above)
Comment (cangencer) [reply hazelcast-jet-core/src/main/resources/hazelcast-jet-default.xml]: It's true by default in the XSD, so it's not needed.

**Pull request merged**